cupy/_statistics/order.py

```diff
@@ -320,21 +330,43 @@ def quantile(a, q, axis=None, out=None, interpolation='linear',
         axis (int or tuple of ints): Along which axis or axes to compute the
             quantiles. The flattened array is used by default.
         out (cupy.ndarray): Output array.
-        interpolation (str): Interpolation method when a quantile lies between
+        method (str): Interpolation method when a quantile lies between
             two data points. ``linear`` interpolation is used by default.
             Supported interpolations are``lower``, ``higher``, ``midpoint``,
             ``nearest`` and ``linear``.
         keepdims (bool): If ``True``, the axis is remained as an axis of
             size one.
+        interpolation (str): Deprecated name for the method keyword argument.
 
     Returns:
         cupy.ndarray: The quantiles of ``a``, along the axis if specified.
 
     .. seealso:: :func:`numpy.quantile`
     """
+    if interpolation is not None:
+        method = _check_interpolation_as_method(
+            method, interpolation, 'quantile')
     if not isinstance(q, cupy.ndarray):
         q = cupy.asarray(q, dtype='d')
     if not _quantile_is_valid(q):  # synchronize
         raise ValueError('Quantiles must be in the range [0, 1]')
-    return _quantile_unchecked(a, q, axis=axis, out=out,
-                               interpolation=interpolation, keepdims=keepdims)
+    return _quantile_unchecked(
+        a, q, axis=axis, out=out, method=method, keepdims=keepdims)
+
+
+# Borrowd from NumPy
+def _check_interpolation_as_method(method, interpolation, fname):
+    # Deprecated NumPy 1.22, 2021-11-08
+    warnings.warn(
+        f"the `interpolation=` argument to {fname} was renamed to "
+        "`method=`, which has additional options.\n"
+        "Users of the modes 'nearest', 'lower', 'higher', or "
+        "'midpoint' are encouraged to review the method they. "
+        "(Deprecated NumPy 1.22)",
+        DeprecationWarning, stacklevel=4)
```

Comment:
If I understand `stacklevel` correctly, it should be at most the number of CuPy functions on the top of the call stack.
```suggestion
        DeprecationWarning, stacklevel=2)
```
But I know what I'm writing means NumPy is wrong, because the logic here is almost the same as NumPy's....

Reply:
Thanks, I paid less attention here. I think NumPy specifies `stacklevel=4` to make the warning refer to percentile/quantile's caller, with additional increment to account for NEP-18 overrides instead of `stacklevel=3` (https://github.com/numpy/numpy/pull/13589). As CuPy does not have NEP-18 dispatching layer, `stacklevel=3` is what it would be. Or `stacklevel=2` is better for us?

Reply:
I missed the dispatching layer, too.

Reply:
`stacklevel=3` looks good.  The warning in #4790 also points user's code.